File: backend/AlphaMind.Api/Entities/Stock.cs
using System.ComponentModel.DataAnnotations;

namespace AlphaMind.Api.Entities;

public class Stock
{
    public int Id { get; set; }

    [Required]
    [MaxLength(30)]
    public string Ticker { get; set; } = string.Empty;

    [Required]
    [MaxLength(200)]
    public string Name { get; set; } = string.Empty;

    [MaxLength(50)]
    public string? Exchange { get; set; }

    public bool IsActive { get; set; }

    public int SortOrder { get; set; }

    public DateTime CreatedAt { get; set; }

    public ICollection<StockNews> NewsItems { get; set; } = [];

    public ICollection<StockAnalysis> Analyses { get; set; } = [];
}

File: frontend/src/types/stock.ts
export type Direction = 'positive' | 'negative' | 'neutral' | 'mixed';

export interface Stock {
  id: number;
  ticker: string;
  name: string;
  exchange?: string | null;
  isActive: boolean;
  sortOrder: number;
}

export interface StockAnalysis {
  id: number;
  stockId: number;
  ticker: string;
  companyName: string;
  impactScore: number;
  confidenceScore: number;
  expectedMove: number;
  direction: Direction;
  summary: string;
  opportunities: string[];
  risks: string[];
  usedNewsCount: number;
  analyzedAt: string;
  createdAt?: string;
}

export interface ApiResult<T> {
  data: T;
  isFallback: boolean;
  error?: string;
}

export interface GroupedAnalysesByDate {
  date: string;
  label: string;
  analyses: StockAnalysis[];
}

export interface AnalysisFilters {
  search: string;
  impactMin: number;
  impactMax: number;
  confidenceMin: number;
  confidenceMax: number;
  moveMin: number;
  moveMax: number;
  direction: Direction | 'all';
}
